lib/signs/utilities/messages.ex

```diff
@@ -14,34 +14,99 @@ defmodule Signs.Utilities.Messages do
+
+        sign_config == :off ->
+          {Content.Message.Empty.new(), Content.Message.Empty.new()}
+
+        sign_config == :headway ->
+          get_headway_or_alert_messages(sign, current_time, alert_status)
+
+        true ->
+          case Signs.Utilities.Predictions.get_messages(predictions, sign) do
+            {%Content.Message.Empty{}, %Content.Message.Empty{}} ->
+              get_headway_or_alert_messages(sign, current_time, alert_status)
+
+            {top_message, %Content.Message.Empty{}} ->
+              {top_message,
+               get_paging_headway_or_alert_messages(sign, current_time, alert_status, :bottom)}
+
+            {%Content.Message.Empty{}, bottom_message} ->
+              if match?(
+                   %Content.Message.Predictions{station_code: "RJFK", zone: "m"},
+                   bottom_message
+                 ) do
+                jfk_umass_headway_paging(bottom_message, sign, current_time, alert_status)
+              else
+                {get_paging_headway_or_alert_messages(sign, current_time, alert_status, :top),
+                 bottom_message}
+              end
+
+            messages ->
+              messages
+          end
+      end
+
+    flip? = flip?(messages)
+
+    if flip?,
+      do: do_flip(messages),
+      else: messages
+  end
 
-      sign_config == :headway ->
-        get_headway_or_alert_messages(sign, current_time, alert_status)
+  defp flip?(messages) do
+    case messages do
+      {%Content.Message.Headways.Paging{}, _} ->
+        true
 
-      true ->
-        case Signs.Utilities.Predictions.get_messages(predictions, sign) do
-          {%Content.Message.Empty{}, %Content.Message.Empty{}} ->
-            get_headway_or_alert_messages(sign, current_time, alert_status)
+      {%Content.Message.Empty{}, _} ->
+        true
 
-          {top_message, %Content.Message.Empty{}} ->
-            {top_message,
-             get_paging_headway_or_alert_messages(sign, current_time, alert_status, :bottom)}
+      _ ->
+        false
+    end
+  end
 
-          {%Content.Message.Empty{}, bottom_message} ->
-            {bottom_message,
-             get_paging_headway_or_alert_messages(sign, current_time, alert_status, :top)}
+  defp do_flip({top, bottom}) do
+    {bottom, top}
+  end
 
-          messages ->
-            messages
-        end
+  # Handles special case when JFK/UMass SB is on headways but NB is on platform prediction
+  defp jfk_umass_headway_paging(prediction, sign, current_time, alert_status) do
+    case get_paging_headway_or_alert_messages(
+           sign,
+           current_time,
+           alert_status,
+           :top
+         ) do
+      %Content.Message.Headways.Paging{destination: destination, range: range} ->
+        {%Content.Message.GenericPaging{
+           messages: [
+             # Make zone nil in order to prevent the usual paging platform message
+             %{prediction | zone: nil},
```

Comment:
With this prediction message nested inside the generic paging message, the `countup?` logic won't trigger as usual. We need to expand that logic if we want it to keep working for this case.

Reply:
Ah great catch, thank you!

Reply:
@panentheos just pushed some changes to expand the countup logic. Mind taking another quick pass? 🙏

Reply:
Ohh hm yes, I think that's correct. We can't match up the predictions when we're transitioning between the two formats and would potentially miss a count-up.

Alternatively, I think we could get around having to do a more substantial refactor by mapping the `PlatformPredictionBottom` struct to a regular `Prediction` struct before calling `countup?` again. I think it should be a safe assumption that the destinations are the same (`:alewife`) since this is such a specific case, or we could also just pass along the `destination` when we create the `PlatformPredictionBottom` struct. We already pass along the minutes to the bottom so we would have all the pieces handy for the countup logic to do its thing.

Reply:
Yeah, I guess that works, as long as we call out with a comment that it's tailored for the special case.